src/main/java/solution/BacktrackingSolution.java

```diff
@@ -0,0 +1,42 @@
+package solution;
+
+import java.awt.*;
+import java.util.List;
+
+class BacktrackingSolution implements Solution {
+
+    private BacktrackingSolution() {
```

Comment:
Very good - no state, just static methods => let's prevent users from creating instances. Just a tip - you can go one step further and throw an exception there to make sure no one will overpass your private constructor via a reflection. But we are talking about how many obstacles you create in order to prevent people from doing bad things.

Reply:
The name of the class is a bit misleading - `BacktrackingSolution` - it makes me think about something that actually has some state. How about e.g. a `BacktrackingSolver`?